### PR TITLE
Fixed #5825

### LINKED ISSUE #5825: Add two english books under Algorithms & Data Structures and Competitive Programming subjects.
Add two books, [Introduction to Algorithms, Third Edition](https://web.iiit.ac.in/~pratik.kamble/storage/Algorithms/Cormen_Algorithms_3rd.pdf) in Algorithms & Data Structures category and [Programming Challenges](http://acm.cs.buap.mx/downloads/Programming_Challenges.pdf) in Competitive Programming category in /books/free-programming-books-subjects.md file.

The above two books are considered the best books for competitive coding.
https://www.geeksforgeeks.org/5-best-books-for-competitive-programming/

### PR DESCRIPTION
**Pull request by AR10X**
## What does this PR do?
Add resource(s) 

### Why is this valuable (or not)?
These books are the best books for competitive programming and DS&algo.

### How do we know it's really free?
Free PDF download links by IIIT Hyderabad and ACM - Universidad Autonoma de Puebla.

### For book lists, is it a book? For course lists, is it a course? etc.
book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Elementary Algorithms](https://github.com/liuxinyu95/AlgoXY) - Larry LIU Xinyu
 * [Foundations of Computer Science](http://infolab.stanford.edu/~ullman/focs.html) - Al Aho and Jeff Ullman
 * [Geometry Algorithms](http://geomalgorithms.com) - Dan Sunday
+* [Introduction to Algorithms, Third Edition](https://web.iiit.ac.in/~pratik.kamble/storage/Algorithms/Cormen_Algorithms_3rd.pdf) - Thomas H. Cormen, Charles E. Leiserson, Ronald L. Rivest, and Clifford Stein (PDF)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:105]: :information_source: I think this is the right site: https://mitpress.mit.edu/books/introduction-algorithms-third-edition

:warning: Not free :question:. Neither of authors is Pratik Kamble. That a link is public on the internet does not mean that its content is free. See [CONTRIBUTING](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#guidelines)

:information_source: Also edition / year goes between parenthesis for ancient versions.

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -154,6 +155,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Competitive Programming, 1st Edition](https://cpbook.net/#CP1details) - Steven Halim [(PDF)](https://www.comp.nus.edu.sg/~stevenha/myteaching/competitive_programming/cp1.pdf)
 * [Competitive Programming, 2nd Edition](https://cpbook.net/#CP2details) - Steven Halim [(PDF)](https://www.comp.nus.edu.sg/~stevenha/myteaching/competitive_programming/cp2.pdf)
 * [Principles of Algorithmic Problem Solving](http://www.csc.kth.se/~jsannemo/slask/main.pdf) - Johan Sannemo (PDF)
+* [Programming Challenges](http://acm.cs.buap.mx/downloads/Programming_Challenges.pdf) - Steven Skiena (PDF)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:158]: Also not free!

See author website: https://www3.cs.stonybrook.edu/~skiena/

**Comment by davorpa** [comment]
### :warning: NOT FREE

**Take care about DMCA**. That a link is public on the internet does not mean that its content is free. See [CONTRIBUTING guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#guidelines)
>  - we don't accept files hosted on google drive, dropbox, mega, scribd, issuu and other similar file upload platforms
>  - use the link with the most authoritative source (meaning the author's website is better than the editor's website, which is better than a third party website)
>    - no file hosting services (this includes (but is not limited to) Dropbox and Google Drive links)

We only accept contents with free licenses like CC (giving creator attributions).

File: books/free-programming-books-subjects.md (hunk 1 above)
Comment (AR10X) [reply books/free-programming-books-subjects.md:105]: Ok, I will take note of this 👍  😄

**Pull request closed without merging**